llvm/lib/Target/X86/X86RegisterInfo.cpp

```diff
@@ -45,6 +45,12 @@ static cl::opt<bool>
 EnableBasePointer("x86-use-base-pointer", cl::Hidden, cl::init(true),
           cl::desc("Enable use of a base pointer for complex stack frames"));
 
+static cl::opt<bool>
+    DisableRegAllocHints("x86-disable-regalloc-hints", cl::Hidden,
+                         cl::init(false),
+                         cl::desc("Disable two address hints for register "
+                                  "allocation"));
```

Comment:
The option name is too generic.

Maybe x86-disable-regalloc-hints-for-ndd

Reply:
c788607